### ISSUE TITLE
[FR] Voltronic /Axpert Invertors Bluetooth support

### ISSUE THREAD

**Issue by Saentist** [jblance/mpp-solar#293]
Is there any information how to read status over Bluetooth in Voltronics / Axpert and clones
All they are controlled by WatchPower app on android
Mostly all Inverters with this control panel.
![image](https://user-images.githubusercontent.com/16975883/210156495-96d0219c-03bf-4e80-8543-41abee31c64e.png) 
![image](https://user-images.githubusercontent.com/16975883/210156692-ee8e45c6-cfb9-462b-bf70-7fb01feae38e.png)

**Comment by jblance** [comment]
Do you have this setup working (with the app)?
Have you logged the bluetooth traffic from the app?

**Comment by Saentist** [comment]
> Do you have this setup working (with the app)? Have you logged the bluetooth traffic from the app?

Yes I have Bluetooth compatible inverter.
But for second part accept any suggestions how to do it.
I can connect to inverter with standard PIN 123456 under Linux/Windows, 
but no COM port emulation is enabled on it as on phones.

**Comment by jblance** [comment]
Yep thats the way - then post the log and/or try to decode it yourself 
It takes a lot of trial and error to reverse engineer the ble log, but I am building a new battery at the moment and have a similar inverter, so will take a look once i get it running

**Comment by Saentist** [comment]
[btsnoop_hci(1).zip](https://github.com/jblance/mpp-solar/files/10371745/btsnoop_hci.1.zip)

Its not so human readable, but `ID55355535553555` is inverter

**Comment by jblance** [comment]
Great. Did you do anything in the app or is it just receiving data?
Can you grab a screen shot of the data

Ps you need wireshark to be able to read it and even then it takes a lot of
guess work

**Comment by Saentist** [comment]
I try to analyse via Wireshark but no success yet not see values.
capture is made with very low spec android 6 phone (Alcatel Pixi 4)
Will try to make new capture with screen recording, 
to be more easy to compare actions and responses.

**Comment by Saentist** [comment]
there is some long repeating string in peyload with lenght 671
````
806001060002428000000000059cbd357d0000000000000000007776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db9cbd357d0000000000000000007776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db9cbd357d0000000000000000007776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db9cbd357d0000000000000000007776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db9cbd357d0000000000000000007776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db7776db6eedb6dbbbb6db776db6ddddb6dbbb6db6eeedb6dddb6db
````

guessing by repeating logic can be something as this
```
806001060002428000000000059
cbd357d000000000000000000
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db9
cbd357d000000000000000000
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db9
cbd357d000000000000000000
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db9
cbd357d000000000000000000
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db9
cbd357d000000000000000000
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
7776db6eedb6dbbbb6db
776db6ddddb6dbbb6db6eeedb6dddb6db
```
repeating segments aka possibly not changed setting from report

first 8 symbols are growing ... guess time or timer.

**Comment by Saentist** [comment]
one more log with video for comparsion
[hci_snoop20230109214525.zip](https://github.com/jblance/mpp-solar/files/10376606/hci_snoop20230109214525.zip)

https://user-images.githubusercontent.com/16975883/211401729-532a9128-f45e-4b7a-b061-9992afe42156.mp4

**Comment by jblance** [comment]
it looks like an encrypted setup
![image](https://user-images.githubusercontent.com/1266998/211423328-d8488c57-6ffe-4bb2-bcda-08b71a974da5.png)
but some info is clearly visible in the trace, eg product info in frame 396
![image](https://user-images.githubusercontent.com/1266998/211423521-d2fd3c6a-2b1d-4be4-9f90-d49d2c9fd796.png)
![image](https://user-images.githubusercontent.com/1266998/211423597-7230d245-8adc-40fc-96e5-c3a93138e17d.png)

**Comment by Saentist** [comment]
There is a encryption key after connecting.

**Comment by jblance** [comment]
response on handle 0x001d might have basic info
![image](https://user-images.githubusercontent.com/1266998/211426342-29f8012a-ccda-41c7-9c5c-ecae2fc5bf75.png)
![image](https://user-images.githubusercontent.com/1266998/211426484-17d5590c-446c-4a27-a5c3-0fc792b5a50f.png)
![image](https://user-images.githubusercontent.com/1266998/211426505-c07db0e8-a41e-434c-b4ba-8905c2fba079.png)
![image](https://user-images.githubusercontent.com/1266998/211426823-fdadf491-d1a2-4949-812f-7f6a5920c594.png)

**Comment by jblance** [comment]
> There is a encryption key after connecting.

yes, but the encryption will make connecting with python harder

**Comment by Saentist** [comment]
> ![image](https://user-images.githubusercontent.com/1266998/211427739-7d43d1c0-04ef-474a-aca0-4d870d9ed945.png)

No battery connected to inverter.

**Comment by jblance** [comment]
> No battery connected to inverter.

yes, just trying to see the same values (after decode) in the log file

**Comment by Saentist** [comment]
first log have PV voltage, MPPT make it flat ~120Vdc

**Comment by jblance** [comment]
handle 0x0023 rated info
![image](https://user-images.githubusercontent.com/1266998/211433125-5b1c377a-880f-404c-abfd-98a08c05a48c.png)
![image](https://user-images.githubusercontent.com/1266998/211433167-9346a5d6-da7f-4e19-ba72-f60f1ed09c8c.png)
handle 0x0023
`fc08d900fc08f401d90088138813e00101000000`
```
fc08 2300
d900 217
fc08 2300
f401 500
d900 217
8813 5000
8813 5000
e001 480
0100 1
0000 0
```

**Comment by Saentist** [comment]
0x0023 inverter rated specs
0x0017 firmware versions
0x001a serial number
0x001d ???
0x0020 ???
0x0026 ???
0x0029 ??? only zeros
0x002c ???
0x002f ???
0x0035 ???
0x0038 ??? * small changes 
0x003b ??? *
0x003e ??? zeros
0x0042 ???
0x0045 ???
0x0048 ??? 
0x004b ???

* respond on value changing in settings.

**Comment by Saentist** [comment]
I try some ble scan tool and this is strange results.
![Screenshot_20230111-232045](https://user-images.githubusercontent.com/16975883/211926914-4ac39962-1100-4b15-af12-5d17c7d0e9af.png)

https://user-images.githubusercontent.com/16975883/211927065-7b1ca994-561f-4f12-b960-22e0e9e3afcc.mp4

[hci_snoop20230109214525 (2).zip](https://github.com/jblance/mpp-solar/files/10396293/hci_snoop20230109214525.2.zip)
[hci_snoop20230109214525 (2)_.zip](https://github.com/jblance/mpp-solar/files/10396296/hci_snoop20230109214525.2._.zip)

sniffing is larger and in two parts

Interesting if inverter s paired via android /not app/ it cannot be scanned

**Comment by jblance** [comment]
> Interesting if inverter s paired via android /not app/ it cannot be scanned

Wonder if this the encryption? 

I thought the inverter I just got was one of the ble ones, turns out it's just wifi
![Uploading 20230112_114942.jpg…]()

**Comment by jblance** [comment]
On the git repo there is a ble-scan.py can you try running that somewhere in range of the inverter (you'll need `pip install bleak` if you don't have it)

**Comment by Saentist** [comment]
Yes will try

----
TI recommend this tool
https://www.ti.com/lit/an/swpa234/swpa234.pdf?ts=1673270553469
https://github.com/Hari-Nagalla/BT-HCI-Logging

----
Wondering how many time will take to "close source payed monitoring" from "free software" to monetize your code
Enable donations in repo ;)

**Comment by jblance** [comment]
The log capture is the easy part, the hard bit is figuring out what the data corresponds to.

I don't think there's money in this code. Helpful for some maybe but that about it

**Comment by Saentist** [comment]
> ble-scan.py

Can you post link to this file?

**Comment by jblance** [comment]
https://github.com/jblance/mpp-solar/blob/master/ble-scan.py

**Comment by DonDavici** [comment]
Maybe this helps you.

https://github.com/DonDavici/OutbackSPC/blob/main/outbackbt.py

**Comment by Saentist** [comment]
@jblance @DonDavici 
can we have some test version for testing

**Comment by Saentist** [comment]
https://github.com/syssi/esphome-votronic/blob/main/votronic_bluetooth.md
Some more code examples.
